lib/helpers/signed-aws-es-fetch.js

```diff
@@ -12,12 +14,80 @@ const defaultAwsKeys = {
 	tokenKeys: ['ES_AWS_SESSION_TOKEN', 'AWS_SESSION_TOKEN']
 };
 
-module.exports = async function (url, opts, creds) {
-	opts = opts || {};
-	url = await resolveUrlAndHost(url);
-	return signedFetch(url, opts, creds);
+// We capture the group before the extension es.amazonaws.com
+const RE_AWS_ES_REGION = /\.(\w{2}-\w*?-\d)\.es\.amazonaws\.com/;
+const defaultRegion = 'eu-west-1';
+
+module.exports = async function (url, opts = {}, creds = {}) {
+	creds = setAwsCredentials(creds);
+	return getSignedRequest(url, opts, creds);
 };
 
+function setAwsCredentials(creds) {
```

Comment:
**question:** I was quite confused about how the whole file has changed, but I think maybe it's just that you have reorganised the file? It makes it quite difficult to understand what's actually changed without reorganising the file myself locally and diffing. How annoying would it be to change the order of the file back and commit it separately?

Reply:
I just sorted it out to follow a logical order and to be easier to understand. Apparently, it wasn't a good idea.

Reply:
It's basically the same code for some URL tools, I split the code to make it easier to check.

Reply:
Heya, sorry I think there was a miscommunication here. I guess what I meant was that the purpose of this PR is to upgrade a dependency, however the PR also includes a lot of refactoring which is unrelated to the dependency bump. For example, the methods were reordered and reorganised.

What I really meant is that in order to review things quickly and easily, it's best if a PR contains _only_ the code related to the task at hand. In this case I want to be able to see just the changes which relate to the dependency upgrade rather than the refactoring, because when the code is rewritten and split across some new files it's much harder to go through line by line and check that the logic is still what we expect.

I'm not against refactoring, it's great! However it's better to have it as a separate PR or at least a separate commit so that it's clear what the intention of each unit of work is. Does that make sense? I struggled to know how to type this in a clear way 😅  so if a quick chat on a call would be easier then I'm happy to do that!

Reply:
Well, there are some different functions than the original one because the implementation is not exactly the same.
Anyway, I reordered and renamed them with the original names.
Another PR can be done with proper names and the refactor.